src/sentry/tasks/base.py

```diff
@@ -94,6 +94,14 @@ def instrumented_task(name, stat_suffix=None, silo_mode=None, record_timing=Fals
     def wrapped(func):
         @wraps(func)
         def _wrapped(*args, **kwargs):
+            from sentry import options
+
+            # Use option to control default behavior of queue time monitoring
+            # Value can be dynamically updated, which is why the evaluation happens during function run-time
+            record_queue_wait_time = record_timing or options.get(
+                "sentry-metrics.monitor-queue-time"
+            )
```

Comment:
Hmm, should I wrap this around a try/catch to contain the blast radius? If any exception doesn't get caught when thrown from the stack all the celery tasks could get effected

Reply:
Yeah might be worth doing so just to be extra cautious

Reply:
done